eisbuk-admin/src/store/selectors/customers.ts

```diff
@@ -22,3 +23,41 @@ export const getCustomersList = (state: LocalStore): Customer[] => {
   const customersInStore = getCustomersRecord(state);
   return Object.values(customersInStore);
 };
+
+/**
+ * Get a list of all the customers whose birthdays are today
+ * (not an actual firestore query but rather synced local store entry)
+ * @param state Local Redux Store
+ * @returns list of customers grouped by birthday
+ */
+export const getCustomersWithBirthday = (
+  state: LocalStore
+): CustomerBirthday => {
+  const customersInStore = getCustomersRecord(state);
+
+  // find index of birthday thats >= todays date to slice at that and concat the rest at the end
+  const sortedCustomers = Object.values(customersInStore).sort((a, b) =>
+    DateTime.fromISO(a.birthday)
+      .set({ year: 2000 })
+      .toString()
+      .localeCompare(
+        DateTime.fromISO(b.birthday).set({ year: 2000 }).toString()
+      )
+  );
+  const index = sortedCustomers.findIndex(
+    (c) =>
+      DateTime.now().startOf("day").set({ year: 2000 }) <=
+      DateTime.fromISO(c.birthday).startOf("day").set({ year: 2000 })
+  );
+  const rearrangedCustomer = sortedCustomers
+    .slice(index === -1 ? 0 : index)
+    .concat(sortedCustomers.splice(0, index));
```

Comment:
Each time you use splice you need to put 5$ in a jar: slice will return the subslice (array) of the passed array, while splice will mutate the passed array (like push/pop).